### DISCUSSION TITLE
Support for archive.org/etree and phish.in

### DISCUSSION THREAD

**Discussion by tube0013** [music-assistant/support#3258] [Feature requests and ideas]
archive.org Live Music Archive (LMA/etree) hosts a large volume of audience recordings of concerts - https://archive.org/details/etree
phish.in does the same for Phish recordings.

relisten.net (<https://github.com/RelistenNet>) integrates both into an online player and apps - bringing this into Music Assistant would be awesome!

**Comment by pljones** [comment]
Just had the same idea.  Currently, to integrate LMA into MA, I have to "cache" (i.e. download to local) copies of LMA content, which isn't really necessary.

However, to work with MA, "collections" would need exploding, then condensing back to "artists".  Then, each live performance would come with multiple instances, which would need handling in some way as it's a layer MA doesn't have.  Relisten just has the same structure as LMA, so doesn't hit those problems.

It would need thinking through carefully.  There's a ton of metadata to consider.

**Reply by pljones** [reply]
Maybe CoPilot could work something up from the sources.

**Comment by Hedda** [comment]
Yes please consider maybe adding a music provider that can stream/play music and audiobooks from the [Internet Archive (”The Internet Achive”)](https://archive.org/).

* https://archive.org

  * https://help.archive.org/help/media-players-a-basic-guide/

Note that audio books on the Internet Archive inlcude LibriVox free audiobooks and "LibriVox" should maybe get its own separate provider?

* https://archive.org/details/audio_bookspoetry

That is, be aware that all the audiobooks on  the Internet Archive are not from LibriVox.

* https://archive.org/details/librivoxaudio
 
  * https://librivox.org/

 - https://github.com/orgs/music-assistant/discussions/4227

Mopidy has backend extension for the that Internet Archive written in Python:

- https://github.com/tkem/mopidy-internetarchive

  - https://pypi.org/project/Mopidy-InternetArchive/

    - https://mopidy.com/ext/internetarchive/
 
Keywords for searching; "Internet Archive" (archive.org InternetArchive) music provider of audio books.

**Comment by OzGav** [comment]
The Internet Archive will be available in the next beta. phish.in is very close to being finished.

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
